common/djangoapps/student/management/commands/retrieve_unsubscribed_emails.py

```diff
@@ -0,0 +1,123 @@
+
+        if not start_date and not end_date:
+            start_date = (datetime.now() - timedelta(days=7)).strftime('%Y-%m-%d')
+            end_date = datetime.now().strftime('%Y-%m-%d')
+
+        try:
+            braze_client = get_braze_client()
+            if braze_client:
+                emails = braze_client.retrieve_unsubscribed_emails(
+                    start_date=start_date,
+                    end_date=end_date,
+                )
+                self.stdout.write(self.style.SUCCESS('Unsubscribed emails retrieved successfully'))
+        except Exception as exc:
+            logger.exception(f'Unable to retrieve unsubscribed emails from Braze due to exception: {exc}')
+            raise CommandError(
+                f'Unable to retrieve unsubscribed emails from Braze due to exception: {exc}'
+            ) from exc
+
+        if emails:
+            try:
+                with tempfile.NamedTemporaryFile(delete=False, mode='wb', suffix='.csv') as csv_file:
+                    csv_file = self._write_csv(csv_file, emails)
+                    csv_file_path = csv_file.name
+            except Exception as exc:
+                logger.exception(f'Error while writing data in CSV file due to exception: {exc}')
+                raise CommandError(
+                    f'Error while writing data in CSV file due to exception: {exc}'
+                ) from exc
+
+            txt_template = 'unsubscribed_emails/email/email_body.txt'
+            html_template = 'unsubscribed_emails/email/email_body.html'
+
+            context = {
+                'start_date': start_date,
+                'end_date': end_date,
+            }
+
+            template = get_template(txt_template)
+            plain_content = template.render(context)
+            template = get_template(html_template)
+            html_content = template.render(context)
+
+            subject = f'Unsubscribed Emails from {start_date} to {end_date}'
+
+            email_msg = EmailMultiAlternatives(
+                subject,
+                plain_content,
+                settings.BRAZE_UNSUBSCRIBED_EMAILS_FROM_EMAIL,
+                settings.BRAZE_UNSUBSCRIBED_EMAILS_RECIPIENT_EMAIL
+            )
+            email_msg.attach_alternative(html_content, 'text/html')
+
+            with open(csv_file_path, 'rb') as file:
+                email_msg.attach(filename='unsubscribed_emails.csv', content=file.read(), mimetype='text/csv')
+
+            try:
+                email_msg.send()
+                logger.info('Unsubscribed emails data sent to your email.')
+            except Exception as exc:
+                logger.exception(f'Failure to send unsubscribed emails data to your email due to exception: {exc}')
+                raise CommandError(
+                    f'Failure to send unsubscribed emails data to your email due to exception: {exc}'
+                ) from exc
+
+        logger.info('No unsubscribed emails found within the specified date range.')
```

Comment:
This is displayed for all cases and shouldn't be. This should only be displayed for when we don't get the emails from braze.

Reply:
Addressed